op-challenger/game/fault/trace/cannon/executor.go

```diff
@@ -131,7 +134,13 @@ func runCmd(ctx context.Context, l log.Logger, binary string, args ...string) er
 	defer stdErr.Close()
 	cmd.Stdout = stdOut
 	cmd.Stderr = stdErr
-	return cmd.Run()
+	start := time.Now()
+	err := cmd.Run()
+	if err != nil {
+		execTime := time.Since(start).Seconds()
+		m.RecordCannonExecutionTime(execTime)
+	}
+	return err
```

Comment:
I'd suggest tracking this metric outside this `runCmd` so we can keep this really minimally focussed on the difficult-to-test actual execution of a process.  Effectively the time it takes for `runCmd` to execute is the cannon execution time as far as op-challenger is concerned.